sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.netstandard2.0.cs

```diff
@@ -115,3 +115,12 @@ public static partial class ResponseExtensions
         public static bool IsError(this Azure.Response response) { throw null; }
     }
 }
+namespace Azure.Messaging
+{
+    public abstract partial class MessageWithMetadata
+    {
+        protected MessageWithMetadata() { }
+        public abstract string ContentType { get; set; }
+        public abstract System.BinaryData Data { get; set; }
```

Comment:
So, are we adding IsReadOnly property?

Reply:
No I don't think it is necessary at this point. If a read-only message is passed to an API that needs to modify it, we have no choice but to throw.

Reply:
We can make the "throwing" a responsibility of the encoder by adding IsReadOnly, but we would still need to throw from the setters in the read-only types, so I didn't see a benefit to adding it.

Reply:
It's a common established pattern in .NET to have abstractions that can be read-write or read-only (e.g. IList). Such abstraction must expose IsReadOnly property such that calling code can avoid exceptions. Reversely, it's not idiomatic to simply throw from such methods.

Reply:
In this case, I don't see how we can avoid throwing an exception if a ServiceBusReceivedMessage is passed into the Encode method.

Reply:
I suppose the consumer can check the IsReadOnly property and elect to not call Encode if IsReadOnly is true, but I don't see this being a common scenario.

Reply:
I suppose this isn't something that we can add later as it needs to be abstract. I will add it given that this is the established convention.